x-pack/plugins/security_solution/public/attack_discovery/use_attack_discovery/index.tsx

```diff
@@ -77,6 +78,8 @@ export const useAttackDiscovery = ({
   // get alerts index pattern and allow lists from the assistant context:
   const { alertsIndexPattern, knowledgeBase, traceOptions } = useAssistantContext();
 
+  const { hasAssistantPrivilege, isAssistantEnabled } = useAssistantAvailability();
```

Comment:
NIT: Since we have set up the capabilities at the links config level now, it is guaranteed that the top-level Attack Discovery page component will never render without the `ai-assistant` privilege or the required license/productType.

The `hasAssistantPrivilege` and `isAssistantEnabled` checks will always be `true`, we could clean them from everywhere in the sub-plugin component tree if we want to simplify the logic. (not the case for connectors read/write or the anonymization sub-feature privileges)

Reply:
> NIT: Since we have set up the capabilities at the links config level now, it is guaranteed that the top-level Attack Discovery page component will never render without the `ai-assistant` privilege or the required license/productType.
> 
> The `hasAssistantPrivilege` and `isAssistantEnabled` checks will always be `true`, we could clean them from everywhere in the sub-plugin component tree if we want to simplify the logic. (not the case for connectors read/write or the anonymization sub-feature privileges)

With [this commit](https://github.com/elastic/kibana/pull/188788/commits/b97fdd16f8dc20ecaeecb0f21dcbd3471e451002) it is possible to for example, configure a custom role that enables Attack discovery, but disables the assistant, as illustrated by the table below:

| Role                                      | License    | Navigation visible | Show upsell | Upsell has actions | View in assistant enabled |
|-------------------------------------------|------------|--------------------|-------------|--------------------|---------------------------|
| `has_attack_discovery_all_assistant_none`     | Basic      | ✅                  | ✅           | ✅                  | ❌                         |
| `has_attack_discovery_all_assistant_none` | Trial      | ✅                  | ❌           | ❌                  | ❌                         |
| `has_attack_discovery_all_assistant_none` | Platinum   | ✅                  | ✅           | ✅                  | ❌                         |
| `has_attack_discovery_all_assistant_none` | Enterprise | ✅                  | ❌           | ❌                  | ❌                         |

The example role above illustrates a scenario where `isAssistantEnabled` will be false. In this scenario, the `View in AI Assistant` buttons must be disabled. As a result, it is not possible to remove the `isAssistantEnabled` checks.